src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -240,28 +240,60 @@ public void editSaleTag(Tag target, Tag editedTag) {
     /**
      * Returns the number of contacts who are associated with the {@code target} tag.
      */
-    public int findByContactTag(Tag target) {
-        return (int) persons.asUnmodifiableObservableList()
-                .stream()
-                .filter(p -> p.getTags().contains(target)).count();
+    public String findByContactTag(Tag target) {
+        StringBuilder output = new StringBuilder();
+        int count = 0;
+        for (Person p : persons) {
+            if (p.getTags().contains(target)) {
+                output.append(String.format("%d. %s\n", count++ + 1, p.toString()));
```

Comment:
Does using `++count` instead work? Same for below

Reply:
Thanks!